src/tests/CalculationLogic.test.ts

```diff
@@ -351,3 +352,67 @@ describe("getPhase", () => {
     expect(getPhase(28, 5, 21)).toEqual(phases.luteal);
   });
 });
+
+describe("getNewCyclesHistory", () => {
+  test("periodDays array is empty", () => {
+    // @ts-expect-error mocked `t` method
+    jest.spyOn(i18n, "t").mockImplementation((key) => key);
+    expect(getNewCyclesHistory([])).toEqual([]);
+  });
+
+  test("periodDays array has a clear ranges", () => {
+    // @ts-expect-error mocked `t` method
+    jest.spyOn(i18n, "t").mockImplementation((key) => key);
+
+    const periodDays = [
+      "2023-08-04",
```

Comment:
Looks like easy to automate

Reply:
For tests it doesn t matter.

Reply:
Ok, I thought those dates would be a side effect for the whole test